x-pack/plugins/saved_objects_tagging/public/components/edition_modal/create_or_edit_modal.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useCallback, useMemo } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiColorPicker,
+  EuiTextArea,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { TagAttributes, TagValidation, validateTagColor } from '../../../common';
+import { TagBadge } from '../../components';
+import { getRandomColor } from './utils';
+
+interface CreateOrEditModalProps {
+  onClose: () => void;
+  onSubmit: () => Promise<void>;
+  mode: 'create' | 'edit';
+  tag: TagAttributes;
+  validation: TagValidation;
+  setField: <T extends keyof TagAttributes>(field: T) => (value: TagAttributes[T]) => void;
+}
+
+export const CreateOrEditModal: FC<CreateOrEditModalProps> = ({
+  onClose,
+  onSubmit,
+  validation,
+  setField,
+  tag,
+  mode,
+}) => {
+  const [submitting, setSubmitting] = useState<boolean>(false);
+
+  // we don't want this value to change when the user edit the name.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialName = useMemo(() => tag.name, []);
+
+  const setName = useMemo(() => setField('name'), [setField]);
+  const setColor = useMemo(() => setField('color'), [setField]);
+  const setDescription = useMemo(() => setField('description'), [setField]);
+
+  const isEdit = useMemo(() => mode === 'edit', [mode]);
+
+  const previewTag: TagAttributes = useMemo(() => {
+    return {
+      ...tag,
+      name: tag.name || 'tag',
+      color: validateTagColor(tag.color) ? '#000000' : tag.color,
+    };
+  }, [tag]);
+
+  const onFormSubmit = useCallback(async () => {
+    if (!validation.valid) {
+      return;
+    }
+
+    setSubmitting(true);
+    await onSubmit();
+    setSubmitting(false);
+  }, [validation, onSubmit]);
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {isEdit ? (
+            <FormattedMessage
+              id="xpack.savedObjectsTagging.management.editModal.title"
+              defaultMessage="Edit '{name}' tag"
+              values={{
+                name: initialName,
+              }}
+            />
+          ) : (
+            <FormattedMessage
+              id="xpack.savedObjectsTagging.management.createModal.title"
+              defaultMessage="Create tag"
+            />
+          )}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiForm data-test-subj="tagModalForm" role="form">
```

Comment:
Prefer the native HTML component if possible

```suggestion
        <EuiForm data-test-subj="tagModalForm" component="form">
```